### ISSUE TITLE
Bind variable (parameter) injection issue

### ISSUE THREAD

**Issue by vprcic** [dbeaver/dbeaver#4844]
I'm a (relatively) new user to DBeaver and am generally quite happy with it. I mostly use it with Oracle DB. However, I must say that I was a bit surprised with the way it handles bind variables. I'm aware that the bind variable behavior was revised and refactored in version 3.8 so it doesn't support native support provided by the drivers. This is clearly stated in issue #1010. There, the user @serge-rider asks the original issue creator what problem they see with the value injection approach.

One real problem is inconsistent query execution time. I started using DBeaver to write queries for development purposes. Later, I started getting complaints from users about the execution time of certain queries. But when I tried recreating the problem and test them in DBeaver, I got wildly different results, where all the queries executed much, much faster. This is both a problem of user experience, as well as one of those pesky but-it-works-jus- fine-on-MY-system problems.

The problem, of course, was the fact that the Oracle optimizer takes different approaches when executing the query with bind variables (which happens within my application) then when executing the injected version (through DBeaver). This, effectively, (at least for me) renders DBeaver unusable for any kind of performance testing, because all the results I get are completely inconsistent with the performance I'll get once the same query is used within my application.

I don't need any fancy "expression" bind variables, which are used as the justification of the injection approach.

So, to sum it up, I'd really like to see the ability to have native-supported bind variables that don't actually change the query that is sent to the DB for execution. I've been a user of various DB tools over the years, and most of them make a distinction between the "injection" bind variables and the native bind variables. This is something I'd really like to see in DBeaver as well.

**Comment by serge-rider** [comment]
Good point. However it is not very clear how to make this obvious for users. Taking into account that most users don't really understand the difference (especially if they are not programmers).
Potentially we can have just one preference option which will determine the behavior. But this won't work for many users because they heavily rely on current behavior (using parameters as a form of macroses - in some cases it is extremely useful).

So, we can make some special way to execute queries (in addition to existing 4 ways). It will work exactly as "Execute statement" but with dynamic parameters binding.
WDYT?

**Comment by brboLikus** [comment]
Ideally, I would suggest two different parameter declarations. Taking example from an alternative SQL querying software, prefixing the parameter name with a colon would imply the native binding, whereas prefixing it with an ampersand (or the ${...}) syntax would imply the "expression"/search-and-replace/inject binding.

However, I most definitely understand the need to not confuse users with changing the known and well-established behavior.

So, I would prefer to have a checkbox (unchecked by default) to "Enable native parameter binding", which would override the now-default injection binding.

What I would not like to see is a new syntax for this kind of execution, because being able to copy-paste between application code and DBeaver is, for me, a must-have.

As to regarding the approach of another way of executing queries, I think this would lead to some confusion as well. In my opinion, most users exclusively and consistently use one or the other binding approach. Having it as an option would allow me to setup my work-space environment (as well as all the environments of people I supervise) and not worry about ways of executing. For me, there is only one way, "the" way to execute a query with bind parameters, and that is the native way...

Thank you for taking my rant in consideration.
